docs/documentation.rst

```diff
@@ -1,46 +1,119 @@
 =============
 Documentation
 =============
+This documentation is generated and published at
+`Read the Docs`_ whenever the master branch is updated.
+
+GitHub can render our ``.rst`` documents as ReStructuredText_, which is
+close enough to Sphinx_ for most code reviews, without features like links
+between documents.
+
+It is occasionally necessary to generate the documentation locally. It is
+easiest to do this with a virtualenv_ on the host system, using Docker only to
+regenerate the MDN Sphinx template.  If you are not comfortable with that style
+of development, it can be done entirely in Docker using ``docker-compose``.
+
+.. _`Read the Docs`: https://kuma.readthedocs.io/en/latest/
+.. _ReStructuredText: https://en.wikipedia.org/wiki/ReStructuredText
+.. _Sphinx: https://en.wikipedia.org/wiki/Sphinx_(documentation_generator)
+.. _virtualenv: https://virtualenv.pypa.io/en/stable/
 
 Generating Documentation
 ------------------------
-This documentation is generated and published at
-`Read the Docs`_ whenever the master branch is updated.
+Sphinx uses a ``Makefile`` in the ``docs`` subfolder to build documentation in
+several formats.  MDN only uses the HTML format, and the generated document
+index is at ``docs/_build/html/index.html``.
 
-To generate locally, install the packages in ``requirements/docs.txt``
-(inside the development VM or in a new `virtualenv`_ on the host system),
-and run::
+To generate the documentation in a virtualenv on the host machine, first
+install the requirements::
+
+    pip install -r requirements/docs.txt
+
+Then switch to the ``docs`` folder to use the ``Makefile``::
 
     cd docs
     make html
+    python -m webbrowser file://${PWD}/_build/html/index.html
+
 
-This will generate the documents with the index at
-``docs/_build/html/index.html``.
+To generate the documentation with Docker::
 
+    docker-compose run --rm web sh -c "\
```

Comment:
This creates the virtualenv inside my kuma working directory, which causes more problems (clean-up, etc.)I can't create the virtualenv in the root folder, because I'm not root anymore.

This works for me:
```
docker-compose run --rm --user $(id -u) web sh -c "\
  virtualenv /tmp/.venvs/docs && \
  . /tmp/.venvs/docs/bin/activate && \
  pip install -r /app/requirements/docs.txt && \
  pip install -e /app/sphinx-theme && \
  cd /app/docs && \
  make html"
```

Please give that a try and report back.